core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournalSnapshot.java

```diff
@@ -0,0 +1,172 @@
+  }
+
+  /**
+   * @return the logs sorted by the end sequence number increasingly
+   */
+  List<UfsJournalFile> getLogs() {
+    return mLogs;
+  }
+
+  /**
+   * @return the temporary checkpoints
+   */
+  List<UfsJournalFile> getTemporaryCheckpoints() {
+    return mTemporaryCheckpoints;
+  }
+
+  /**
+   * Creates a snapshot of the journal.
+   *
+   * @return the journal snapshot
+   * @throws IOException if any I/O errors occur
+   */
+  static UfsJournalSnapshot getSnapshot(UfsJournal journal) throws IOException {
+    // Checkpoints.
+    List<UfsJournalFile> checkpoints = new ArrayList<>();
+    UnderFileStatus[] statuses = journal.getUfs().listStatus(journal.getCheckpointDir().toString());
+    if (statuses != null) {
+      for (UnderFileStatus status : statuses) {
+        UfsJournalFile file = UfsJournalFile.decodeCheckpointFile(journal, status.getName());
+        if (file != null) {
+          checkpoints.add(file);
+        }
+      }
+      Collections.sort(checkpoints);
+    }
+
+    List<UfsJournalFile> logs = new ArrayList<>();
+    statuses = journal.getUfs().listStatus(journal.getLogDir().toString());
+    if (statuses != null) {
+      for (UnderFileStatus status : statuses) {
+        UfsJournalFile file = UfsJournalFile.decodeLogFile(journal, status.getName());
+        if (file != null) {
+          logs.add(file);
+        }
+      }
+      Collections.sort(logs);
+    }
+
+    List<UfsJournalFile> tmpCheckpoints = new ArrayList<>();
+    statuses = journal.getUfs().listStatus(journal.getTmpDir().toString());
+    if (statuses != null) {
+      for (UnderFileStatus status : statuses) {
+        tmpCheckpoints.add(UfsJournalFile.decodeTemporaryCheckpointFile(journal, status.getName()));
+      }
+    }
+
+    return new UfsJournalSnapshot(checkpoints, logs, tmpCheckpoints);
+  }
+
+  /**
+   * Gets the current log (the incomplete log) that is being written to.
+   *
+   * @return the current log
+   * @throws IOException if any I/O errors occur
+   */
+  static UfsJournalFile getCurrentLog(UfsJournal journal) throws IOException {
+    List<UfsJournalFile> logs = new ArrayList<>();
+    UnderFileStatus[] statuses = journal.getUfs().listStatus(journal.getLogDir().toString());
+    if (statuses != null) {
+      for (UnderFileStatus status : statuses) {
+        UfsJournalFile file = UfsJournalFile.decodeLogFile(journal, status.getName());
+        if (file != null) {
+          logs.add(file);
+        }
+      }
+      if (!logs.isEmpty()) {
+        UfsJournalFile file = Collections.max(logs);
+        if (file.isIncompleteLog()) {
+          return file;
+        }
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Gets the first journal log sequence number that is not yet checkpointed.
+   *
+   * @return the first journal log sequence number that is not yet checkpointed
+   * @throws IOException if any I/O errors occur
+   */
+  static long getNextLogSequenceToCheckpoint(UfsJournal journal) throws IOException {
```

Comment:
why is this a static method of UfsJournalSnapshot as opposed to a member method of UfsJournal?